posts/epi-community-contrib/index.qmd

```diff
@@ -0,0 +1,77 @@
+
+Users can now specify the proportion of presymptomatic transmission rather than having to understand the skew normal parameterisation used by the simulation model, making it easier to get started with the package for new users.
+
+Lastly on user-facing changes, the naming and style of function arguments has been standardised for consistent use of [snakecase](https://en.wikipedia.org/wiki/Snake_case) style and abbreviations.  
+
+### Documentation
+
+Function documentation already used {roxygen2}, but did not make use of inheritance or comprehensively document the function output or usage. We used `@inheritParams` from {roxygen2} to deduplicate, added `@return` documentation to all functions. We also improved the function argument documentation by following a structure of: `<type>: description`, for example:
+
+```r
+@param sim a `list` with class `<ringbp_sim_opts>`: the simulation control
+  options for the \pkg{ringbp} model, returned by [sim_opts()]
+```
+
+Exported functions now have informative examples (`@examples`) to showcase how the functions should be used. Function examples now always run (removing `\dontrun{}`) to catch any errors.
+
+The [{roxyglobals}](https://github.com/anthonynorth/roxyglobals) package has been added to automate the management of global variables with the use of the `@autoglobal` tag.
+
+Vignettes are useful long-form package documentation. Thus far we've added one vignette to the package and plan to add more where relevant.
+
+### Bug fixes
+
+Perhaps more important that any of the software best practices and user interface is the correctness of the code. In our developments we've uncovered a few bugs in the previous version of {ringbp}. The timing of quarantining infected individuals, sampling from the onset-to-isolation distribution, and calculating the generation time from the incubation period have all been identified and fixed.
+
+### Testing
+
+- simulation correctness regression (snapshot) testing
+
+### Miscellaneous
+
+There are various other changes in {ringbp} from our work. Examples include: input checking, not specifying erroneous function defaults, updating the package website, and functions that return `data.table` objects no longer [return silently](https://cran.r-project.org/web/packages/data.table/vignettes/datatable-faq.html#sec:why-do-i-have-to-type-dt-sometimes-twice-after-using-to-print-the-result-to-console). Mentioned in the introduction, model performance has been incrementally improved, but we've not focused on this aspect, and the package will benefit from time spent focusing on this in the future; especially if the set and complexity of non-pharmaceutical interventions in the model expands.
```

Comment:
```suggestion
There are various other changes in {ringbp} from our work. Examples include: input checking, not specifying erroneous function defaults, updating the package website, and functions that return `data.table` objects no longer [returning silently](https://cran.r-project.org/web/packages/data.table/vignettes/datatable-faq.html#sec:why-do-i-have-to-type-dt-sometimes-twice-after-using-to-print-the-result-to-console). Mentioned in the introduction, model performance has been incrementally improved, but we've not focused on this aspect, and the package will benefit from time spent focusing on this in the future; especially if the set and complexity of non-pharmaceutical interventions in the model expands.
```

Reply:
I've just run a rough benchmarking comparison and the newest version of the package (`min-gen-time` branch on my fork) has approximately the same (if not marginally slower) runtime than the version before I started working on the package (using package version at [`dbf85c1`](https://github.com/epiforecasts/ringbp/commit/dbf85c1846fac9ccd022da317adedaf40d928fa9)). The newer version has smaller memory allocation. Therefore I'll keep the current wording which is ambiguous about "performance".

Reply:
hmm! i guess things fell off w/ some flexibility introductions or when incorporating corrections that entail drawing more samples.